src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,247 @@
+// eslint-disable-next-line
+import { useParams } from 'react-router';
+import { useEffect, useState } from 'react';
+import { Link, useSearchParams } from 'react-router-dom';
+import classNames from 'classnames';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
 
-export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+export const PeoplePage: React.FC<{
+  isError: boolean;
+  setIsError: React.Dispatch<React.SetStateAction<boolean>>;
+  isLoading: boolean;
+  peopleFromServer: Person[] | undefined;
+  errorMessage: string | undefined;
+}> = ({
+  isError, setIsError, isLoading, peopleFromServer, errorMessage,
+}) => {
+  const { slug = '' } = useParams();
+  const [visiblePeople, setVisiblePeople] = useState(peopleFromServer || []);
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  const initialSort = searchParams.get('sort');
+  const [sortUpdate, SetSortUpdate] = useState(initialSort);
+  const [sortBefore, SetSortBefore] = useState(sortUpdate);
+  const initialOrder = searchParams.get('order') || null;
+  const [orderUpdate, SetOrderUpdate] = useState(initialOrder);
+  const sortLane = ['Name', 'Sex', 'Born', 'Died', 'Mother', 'Father'];
```

Comment:
FYI: you can move such constants out of component

Reply:
As on slide upper. No. useState I can use only from within the executable program.

Reply:
And if I move all what you ask me, program not work correctly.